hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -1919,6 +1933,21 @@ public Collection<? extends BlockStoragePolicySpi> getAllStoragePolicies()
       }
       return allPolicies;
     }
+
+    @Override
+    public Path getEnclosingRoot(Path path) throws IOException {
+      InodeTree.ResolveResult<FileSystem> res;
+      try {
+        res = fsState.resolve((path.toString()), true);
+      } catch (FileNotFoundException ex) {
+        throw new NotInMountpointException(path, String.format("getEnclosingRoot - %s", ex.getMessage()));
```

Comment:
add the caught exception as the cause

Reply:
im not totally sure what the ask is here, this is how i've been doing all of them. Are you saying only have the caught exception as the cause without the `getEnclosingRoot - "` part?

Reply:
no, I mean that .initCause() is needed to preserve the entire stack trace. otherwise root causes of problems may get lost.